sql/core/src/main/scala/org/apache/spark/sql/execution/local/LocalNode.scala

```diff
@@ -29,10 +31,18 @@ import org.apache.spark.sql.types.StructType
  * Before consuming the iterator, open function must be called.
  * After consuming the iterator, close function must be called.
  */
-abstract class LocalNode extends TreeNode[LocalNode] {
+abstract class LocalNode(conf: SQLConf) extends TreeNode[LocalNode] with Logging {
+
+  val codegenEnabled: Boolean = conf.codegenEnabled
+
+  val unsafeEnabled: Boolean = conf.unsafeEnabled
```

Comment:
and the `ProjectNode` always uses unsafe projection, should we control that by this config?